controllers/dspipeline_params.go

```diff
@@ -579,6 +586,20 @@ func setResourcesDefault(defaultValue dspa.ResourceRequirements, value **dspa.Re
 	}
 }
 
+func (p *DSPAParams) LoadMlmdCertificates(ctx context.Context, client client.Client, logger logr.Logger) (bool, error) {
+	secret, err := util.GetMlmdGrpcCertificatesSecret(ctx, client, p.Namespace, p.Name)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			return false, nil
+		} else {
+			return false, err
+		}
+	}
+	p.MlmdGrpcCertificateContents = strings.ReplaceAll(string(secret.Data["tls.crt"]), "\n", "\\n")
```

Comment:
am i reading this right? ReplaceAll newline with the same newline seems like it shouldn't do anything

Reply:
It's not the same newline. I'm replacing it with an escaped `\`.